lib/did-io.js

```diff
@@ -402,6 +526,52 @@ api.promises = function(options) {
 })();
 
 /**
+ * Helper function to establish a proof of patience for registering a DID.
+ * @param [options] the options to use:
+ *          [baseUrl] the target endpoint for the request
+ *          [request] instance of the request library.
+ *          [didDocument] the DID document to register.
+ *          [strictSSL] enable/disable strictSSL.
+ * @param callback(err, result) called once the operation completes.
+ */
+function _establishProofOfPatience(options, callback) {
+  var request = options.request;
+  request.post(options.baseUrl, {
+    headers: {'Content-Type': 'application/ld+json'},
+    body: JSON.stringify(options.didDocument),
+    strictSSL: options.strictSSL
+  }, function(err, res) {
+    if(err) {
+      return callback(err);
+    }
+    // success
+    if(res.statusCode < 300) {
+      return callback(null, {registered: true});
+    }
+    // if not a 401 HTTP error code w/proof-of-patience then bail
+    if(!(res.statusCode === 401 &&
+      'retry-after' in res.headers &&
+      'www-authenticate' in res.headers)) {
+      return callback(new BedrockError(
```

Comment:
did-io is not a bedrock lib, this `BedrockError` will cause a parse error. It needs to be changed to a regular `Error`. This also means the meta data like the error name and `httpStatusCode` details can't be captured the same way -- we'll want to include that information in the message and potentially check for it and wrap it in a `BedrockError` where this is called in a bedrock system.